### ISSUE TITLE
AssertionError in assert not torch.isnan(h_prime).any()

### ISSUE THREAD

**Issue by iamlockelightning** [Diego999/pyGAT#11]
I train a GAT network. After 2 epoch I got the AssertionError as follow:
![image](https://user-images.githubusercontent.com/12706469/47295096-704b4700-d641-11e8-93b8-f2e028901480.png)

I wonder why I get **NaN** in this two assertion.
![image](https://user-images.githubusercontent.com/12706469/47295225-b7d1d300-d641-11e8-9949-0925f1d768f0.png)

python version: 3.6
pytorch version: 0.4
os: mac os

Anyone can help? Thanks!

**Comment by diegoantognini** [comment]
Did you try with the non-sparse version ? The sparse version is slightly different (cf PR #8 ). @sh0416 Please could you have a check ?

**Comment by sh0416** [comment]
you must use pytorch 0.4.1 because API for sparse version is different

**Comment by sh0416** [comment]
Also, I found that GAT for sparse version cannot backprop because creation is just copying the value of tensor. I trying to make tosparse custom API to handle this. for now, please use dense version

**Comment by sh0416** [comment]
Also, check the initail value because softmax can make numerical instability.

**Comment by iamlockelightning** [comment]
Thank you so much for your quick and helpful replies!! @Diego999 @sh0416

**Comment by qhd1996** [comment]
This is because you use dropout in input, thus make some rowsum zero.

**Comment by tanjia123456** [comment]
Hello, I have met the same problems, could you please tell me how to solve it?

**Comment by kangleXU** [comment]
@iamlockelightning Hello, solved this problem ?  could you please tell me how to solve it?
